src/Microsoft.AspNetCore.ResponseCaching/ResponseCachingMiddleware.cs

```diff
@@ -211,7 +210,7 @@ public class ResponseCacheMiddleware
                 // Create the cache entry now
                 var response = context.HttpContext.Response;
                 var varyHeaders = new StringValues(response.Headers.GetCommaSeparatedValues(HeaderNames.Vary));
-                var varyQueryKeys = context.HttpContext.Features.Get<IResponseCacheFeature>()?.VaryByQueryKeys ?? StringValues.Empty;
+                var varyQueryKeys = new StringValues(context.HttpContext.Features.Get<IResponseCachingFeature>()?.VaryByQueryKeys);
```

Comment:
Converting to `StringValues` here to take advantage of utilities such as `Equals` and `IsNullOrEmpty`. Could consider duplicating these utilities for `string[]` if the extra allocation of a `StringValues` struct is too expensive.